Server/src/bmp/BMPReader.cpp

```diff
@@ -270,9 +271,31 @@ bool BMPReader::ReadIncomingMsg(BMPListener::ClientInfo *client, MsgBusInterface
             case parseBMP::TYPE_INIT_MSG : { // Initiation Message
                 LOG_INFO("%s: Init message received with length of %u", client->c_ip, pBMP->getBMPLength());
                 pBMP->handleInitMsg(read_fd, r_object);
-
-                // Update the router entry with the details
-                mbus_ptr->update_Router(r_object, mbus_ptr->ROUTER_ACTION_INIT);
+		
+//NAT-PAT feature.
+		char *hash_val=client->c_ip;
+		if(cfg->pat_enabled)
```

Comment:
Would be good to move this to a method since it needs to be called more than once.